src/index.js

```diff
@@ -0,0 +1,88 @@
+import { helpers } from '@eik/common';
+import { join } from 'path';
+import fetch from 'node-fetch';
+
+const isUrl = (value = '') => value.startsWith('http');
+
+const fetchImportMaps = async (urls = []) => {
+    try {
+        const maps = urls.map((map) => fetch(map).then((result) => {
+            if (result.status === 404) {
+                throw new Error('Import map could not be found on server');
+            } else if (result.status >= 400 && result.status < 500) {
+                throw new Error('Server rejected client request');
+            } else if (result.status >= 500) {
+                throw new Error('Server error');
+            }
+            return result.json();
+        }));
+        return await Promise.all(maps);
+    } catch (err) {
+        throw new Error(
+            `Unable to load import map file from server: ${err.message}`,
+        );
+    }
+};
+
+export default class EikNodeClient {
+    constructor({
+        development = false,
+        base = '',
+        path = process.cwd(),
+    } = {}) {
+        this.pDevelopment = development;
+        this.pConfig = {};
+        this.pPath = path;
+        this.pBase = base;
+        this.pMaps = [];
```

Comment:
I’ve been thinking a bit about object properties and how best to separate those that should be public and those that should be private. I don’t think we are ready to be using private class properties on untranspiled code yet so I think probably using symbols is still the best way to go for now. I think these “p” namespaced properties should be made private yea?

Reply:
Yeah. But I'm actually gonna run with private properties here. We are after all on node versions supporting it.